Resources/public/js/views/ez-dashboardblockbasedview.js

```diff
@@ -0,0 +1,217 @@
+                    return a.get('identifier') < b.get('identifier');
+                } else {
+                    // sort by priority (max to 0)
+                    return priorityA > priorityB;
+                }
+            });
+
+            return list;
+        },
+
+        /**
+         * Verifies provided `blocks` attribute value
+         * by checking whether provided values are dashboard block view instances
+         *
+         * @method _checkBlocks
+         * @protected
+         * @param value {Any} new vaue of `blocks` attribute
+         * @return {Object}
+         */
+        _checkBlocks: function (value) {
+            var blocks = {};
+
+            if (typeof value !== 'object' || Array.isArray(value)) {
+                return blocks;
+            }
+
+            Object.keys(value).forEach(function (key) {
+                if (value[key] instanceof Y.eZ.BaseDashboardBlockView) {
+                    blocks[key] = value[key];
+                }
+            });
+
+            return blocks;
+        },
+
+        /**
+         * Adds a block to the dashboard
+         *
+         * @method addBlock
+         * @param view {Y.eZ.BaseDashboardBlockView} dashboard block view instance
+         * @return {Object} the hash of blocks
+         */
+        addBlock: function (view) {
+            var blocks = this.get('blocks'),
+                identifier = view.get('identifier');
+
+            if (!identifier) {
+                return blocks;
+            }
+
+            if (blocks[identifier]) {
+                blocks = this.removeBlock(identifier, true);
+            }
+
+            blocks[identifier] = view;
+
+            this.set('blocks', blocks);
+
+            return blocks;
+        },
+
+        /**
+         * Removes a block from the dashboard
+         *
+         * @method removeBlock
+         * @param identifier {String} block identifier
+         * @param preventUpdating {Boolean} should the `blocks` attribute value be updated?
+         * @return {Object} the hash of blocks
+         */
+        removeBlock: function (identifier, preventUpdating) {
+            var blocks = this.get('blocks'),
+                block = blocks[identifier];
+
+            block.removeTarget(this);
+            block.destroy({remove: true});
+
+            delete blocks[identifier];
+
+            if (!preventUpdating) {
+                this.set('blocks', blocks);
+            }
+
+            return blocks;
+        }
+    }, {
+        ATTRS: {
+            /**
+             * The dashboard title
+             *
+             * @attribute dashboardTitle
+             * @type String
+             * @default 'Dashboard'
+             */
+            dashboardTitle: {
+                value: 'Dashboard'
+            },
+
+            /**
+             * Hash containing information about all blocks in the dashboard
+             *
+             * @attribute blocks
+             * @type Object
```

Comment:
You should store the blocks as an array that would ease the addition of new blocks while not making the removal complicated (just filter the array by identifier).
Also, this attribute should be read only as we want the external developers to use `addView`. That also means we don't need the setter.